### PR TITLE
Updated pdfjs to 4+

### PR DESCRIPTION
**Pull request by nullpointerexceptionkek**
This PR updates PDFJS-DIST dependency 3.x to 4.6.82. PDFjs 4 introduces some breaking changes, including the deprecation of rendertextlayer. 

PDFJS 3.x is currently deprecated and has vulnerabilities.

This PR updates various imports and uses a new method to render the text layer.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -9,6 +9,6 @@
 		"skipLibCheck": true,
 		"sourceMap": true,
 		"strict": true,
-		"moduleResolution": "node16"
+		"module": "preserve",
```
Comment (gtm-nayan) [inline tsconfig.json]: Why this change?

Comment (louisqli) [reply tsconfig.json]: I change some configuration while testing.
I can revert this.

File: src/routes/+layout.svelte [hunk 2]
```diff
@@ -1,5 +1,6 @@
-<script lang="ts">
+<script type="module" lang="ts">
```
Comment (gtm-nayan) [inline src/routes/+layout.svelte]:
```suggestion
<script lang="ts">
```

setcontext doesn't work inside the module script

**Review by gtm-nayan — APPROVED** [review]
Thanks, could you also revert moving the workflow file, then I'll merge and publish the next version.

**Comment by nullpointerexceptionkek** [comment]
Should be reverted
